client/src/main/java/org/evosuite/testcase/TestFactory.java

```diff
@@ -291,27 +294,39 @@ private int doInjection(TestCase test, int position, Class<?> klass, VariableRef
 				test.addStatement(ms, injectPosition++);
 			}
 
-			//then do the non-special fields that need injection
+			// then do the non-special fields that need injection
 			for (Field f : Injector.getGeneralFieldsToInject(target)) {
 
 				/*
-					Very tricky: if we allow to reuse a variable X, it might end up
-					that X is a bounded variable previously created for injection
-					but where the initialization calls have not been added yet to the test.
-					Handling it "properly" would be far too complicated :(
-					So we just avoid reusing existing variables in a recursive call, as
-					anyway we can always rely on FM to "save the day"
+				 * Very tricky: if we allow to reuse a variable X, it might end
+				 * up that X is a bounded variable previously created for
+				 * injection but where the initialization calls have not been
+				 * added yet to the test. Handling it "properly" would be far
+				 * too complicated :( So we just avoid reusing existing
+				 * variables in a recursive call, as anyway we can always rely
+				 * on FM to "save the day"
 				 */
-				boolean reuseVariables  = recursionDepth == 0;
+				boolean reuseVariables = recursionDepth == 0;
 
 				int beforeLength = test.size();
-				VariableReference valueToInject = satisfyParameters(
-						test,
-						ref, // avoid calling methods of bounded variables
-						Arrays.asList((Type) f.getType()),
-						injectPosition,
-						recursionDepth +1,
-						false, true, reuseVariables).get(0);
+				VariableReference valueToInject = satisfyParameters(test, ref, // avoid
+																				// calling
+																				// methods
+																				// of
+																				// bounded
+																				// variables
+						Arrays.asList((Type) f.getType()), null, // Added 'null'
+																	// as
```

Comment:
These comments wrapped over multiple lines are a bit odd - maybe don't reformat the code automatically